### PR TITLE
Fix sphinx doc

### PR DESCRIPTION
**Pull request by hiergaut**
Make sphinx documentation without compilation error

### CODE REVIEW

**Comment by hiergaut** [comment]
waiting for #21 merged

File: .github/workflows/deployDoc.yml [hunk 1]
```diff
@@ -22,12 +22,18 @@ jobs:
     - name: Configure CMake
       run: cmake -B "${{github.workspace}}/build" -DHUB_BUILD_DOC=ON
       
-    - name: Build doxygen doc
-      run: cmake --build ${{github.workspace}}/build --target hub-doc-doxygen
```
Comment (nmellado) [inline .github/workflows/deployDoc.yml:26]: So if I understand well we do not deploy doxygen documentation anymore ?

Comment (hiergaut) [reply .github/workflows/deployDoc.yml:26]: The doxygen documentation is also deployed by sphinx

File: .github/workflows/deployDoc.yml [hunk 2]
```diff
@@ -22,12 +22,18 @@ jobs:
...
+
+    - name: Build sphinx doc
+      run:
+        cmake --build ${{github.workspace}}/build --target hub-doc-sphinx
 
     - name: Deploy doc 🚀
       uses: JamesIves/github-pages-deploy-action@v4
       with:
-        folder: ${{github.workspace}}/build/doc/docs_doxygen/html # The folder the action should deploy.
+        folder: ${{github.workspace}}/doc/sphinx/_build/html/ # The folder the action should deploy.
```
Comment (nmellado) [inline .github/workflows/deployDoc.yml:37]:
```suggestion
        folder: ${{github.workspace}}/build/doc/sphinx/_build/html/ # The folder the action should deploy.
```

Comment (hiergaut) [reply .github/workflows/deployDoc.yml:37]: The compilation of sphinx imply doxygen compilation, because sphinx uses doxygen doc.

Comment (hiergaut) [reply .github/workflows/deployDoc.yml:37]: The sphinx doc is compiled locally. I am going to make compilation in build folder.

**Comment by nmellado** [comment]
Let me know when I can review

**Comment by hiergaut** [comment]
I had some problem (with xml path in server path) to build in build folder, I will fix this probably tomorow.

File: .github/workflows/deployDoc.yml (hunk 2 above)
Comment (hiergaut) [reply .github/workflows/deployDoc.yml]: Done, check the file.

File: doc/sphinx/Doxyfile.in [hunk 3]
```diff
@@ -68,7 +68,8 @@ PROJECT_LOGO           =
 # entered, it will be relative to the location where doxygen was started. If
 # left blank the current directory will be used.
 
-OUTPUT_DIRECTORY       = "_build"
+OUTPUT_DIRECTORY       = _build
+# OUTPUT_DIRECTORY       = @CMAKE_CURRENT_BINARY_DIR@/sphinx/
```
Comment (nmellado) [inline doc/sphinx/Doxyfile.in:72]: Do we need to keep this comment ?

File: doc/sphinx/Doxyfile.in [hunk 4]
```diff
@@ -917,7 +918,8 @@ WARN_LOGFILE           =
 # spaces. See also FILE_PATTERNS and EXTENSION_MAPPING
 # Note: If this tag is empty the current directory is searched.
 
-INPUT                  =  "../../src/" "../../examples/"
+INPUT                  =  "../../src/" "../../examples/" "../../doc/"
+# INPUT       = "@CMAKE_SOURCE_DIR@/src/" "@CMAKE_SOURCE_DIR@/examples/" "@CMAKE_SOURCE_DIR@/doc/"
```
Comment (nmellado) [inline doc/sphinx/Doxyfile.in:922]: Do we need to keep this comment ?

**Review by nmellado — COMMENTED** [review]
Except two minor comments, looks fine !

**Comment by hiergaut** [comment]
I tested the deploy.

File: doc/sphinx/Doxyfile.in (hunk 3 above)
Comment (hiergaut) [reply doc/sphinx/Doxyfile.in:72]: And add todo target in Doxyfile.in to fix this.

Comment (hiergaut) [reply doc/sphinx/Doxyfile.in:72]: You can remove the comment. Will fix later. I will do some issue.

File: doc/sphinx/Doxyfile.in (hunk 4 above)
Comment (hiergaut) [reply doc/sphinx/Doxyfile.in:922]: You can remove the comment. Will fix later. I will do some issue.

**Comment by hiergaut** [comment]
An issue #47 was added to make all build files into build dir.

File: doc/sphinx/Doxyfile.in (hunk 3 above)
Comment (hiergaut) [reply doc/sphinx/Doxyfile.in]: Comment removed.

File: doc/sphinx/Doxyfile.in (hunk 4 above)
Comment (hiergaut) [reply doc/sphinx/Doxyfile.in]: Comment removed.

**Pull request merged**